Problems with Equation 2 (#41)
MrLongbottom/P9-Rapport: sections/appendix/problems.tex

In the end we decided that reverse engineering the \gls{lda} process, was not an effective strategy, and went on to try getting more context to our queries using 'query extension'.

\subsubsection{Duplicates within the dataset}
Within Nordjyske's data, each article is published in several different magazines. 
This results in the dataset consisting of many duplicates. 
To combat this problem, we make a dictionary of articles, so that we only process unique articles.
However when we have preprocessed the Nordjyske data, there are still some duplicates, since articles containing information like football scoreboards, contain different scores before preprocessing but the same information after preprocessing, since we remove numbers from the articles.

\subsubsection{Problems with word value multiplication}
We use \autoref{eq:query_prob} to calculate the probability of a query having been generated by a specfic document. However because all word probabilities $P(w|d)$ are multiplied together, having a value of 0 just one word will make the whole query have a value of 0.

> This has shown to be a problem in several of our baselines (namely \gls{lm} and \gls{tf-idf}), which ruteenly evaluates words to be 0.

Reviewer: hvad skal der stå i stedet for ruteenly ?

Author: ```suggestion
This has shown to be a problem in several of our baselines (namely \gls{lm} and \gls{tf-idf}), often gives words a value of 0.
```

> One of the consequences of this problem is that the algorithms give the same value for all queries containing at least one 0-value word, even when the other words have different values. Queries that contain one 0-value words are also evaluated the same as queries containing more 0-value words.

Reviewer: ```suggestion
One of the consequences of this problem is that the algorithms give the same value for all queries containing at least one 0-value word, even when the other words have different values. 
Queries that contain one 0-value words are also evaluated the same as queries containing more 0-value words.
```

> In order to counteract this problem we have implemented a minimum value when calculating this equation. Each 0 value is replaced with $?$\todo{insert value}. This value is so low that the query will always score higher than queries containing no 0-value words, but it makes it possible to sort queries containing one or more 0-value words.

Reviewer: ```suggestion
To counteract this problem, we have implemented a minimum value when calculating this equation. 
Each 0 value is replaced with $?$\todo{insert value}. 
This value is so low that the query will always score higher than queries containing no 0-value words, but it makes it possible to sort queries containing one or more 0-value words.
```

In the end we decided that reverse engineering the \gls{lda} process, was not an effective strategy, and went on to try getting more context to our queries using 'query extension'.

\subsubsection{Duplicates within the dataset}
Within Nordjyske's data, each article is published in several different magazines. 
This results in the dataset consisting of many duplicates. 
To combat this problem, we make a dictionary of articles, so that we only process unique articles.
However when we have preprocessed the Nordjyske data, there are still some duplicates, since articles containing information like football scoreboards, contain different scores before preprocessing but the same information after preprocessing, since we remove numbers from the articles.

\subsubsection{Problems with word value multiplication}
We use \autoref{eq:query_prob} to calculate the probability of a query having been generated by a specific document. 

> However, because all word probabilities $P(w|d)$ are multiplied together, having a value of 0 just one word will make the whole query have a value of 0.

Reviewer 2: ```suggestion
However, because all word probabilities $P(w|d)$ are multiplied together, having a value of 0 for just one word will make the whole query have a value of 0.
```

> This has shown to be a problem in several of our baselines (namely \gls{lm} and \gls{tf-idf}), which ruteenly evaluates words to be 0.

Author: ```suggestion
This has shown to be a problem in several of our baselines (namely \gls{lm} and \gls{tf-idf}), often gives words a value of 0.
```

One of the consequences of this problem is that the algorithms give the same value for all queries containing at least one 0-value word, even when the other words have different values. 

> Queries that contain one 0-value words are also evaluated the same as queries containing more 0-value words.

Reviewer 2: ```suggestion
Queries that contain one 0-value word are also evaluated the same as queries containing multiple 0-value words.
```

To counteract this problem, we have implemented a minimum value when calculating this equation. 

> Each 0 value is replaced with $?$\todo{insert value}. 

Reviewer 2: ```suggestion
Each 0-value is replaced with $?$\todo{insert value}. 
```

> This value is so low that the query will always score higher than queries containing no 0-value words, but it makes it possible to sort queries containing one or more 0-value words.

Reviewer 2: ```suggestion
This value is so low that the query will always score lower than queries containing no 0-value words, but it makes it possible to sort queries containing one or more 0-value words.
```
Hvis vi her snakker om probability multiplication scoren.
